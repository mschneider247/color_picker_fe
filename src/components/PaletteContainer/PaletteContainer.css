#Section_PaletteContainer {
  display: flex;
}

.colorBox {
  width: 15%;
  height: 200px;
}