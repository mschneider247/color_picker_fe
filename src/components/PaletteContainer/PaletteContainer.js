import React from 'react';
import './PaletteContainer.css'

const PalettesContainer = (props) => {

  let colorBoxes = props.colors.map((color, index) => {
    let style = {
      backgroundColor: color
    }
    return (
      <div key={index} className="colorBox" style={style}></div>
    )
  });

  return(
    <section id="Section_PaletteContainer">
      {colorBoxes}
    </section>
  )
};

export default PalettesContainer;