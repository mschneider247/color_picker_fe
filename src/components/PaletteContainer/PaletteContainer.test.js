import React from 'react';
import { shallow } from 'enzyme';
import PaletteContainer from './PaletteContainer';


describe('PaletteContainer component', () => {
	let wrapper;
	let mockUpdateLockedIndex = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<PaletteContainer  updateLockedIndex={mockUpdateLockedIndex} colors={['','','','','']}/>)
  });

  it('App should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when updateIndex is invoked', () => {
  	let mockEvent = { target: { value: 4 } };
  	wrapper.instance().updateIndex(mockEvent);
  	expect(wrapper.state('lockedIndex')).toEqual(4);
  	expect(mockUpdateLockedIndex).toHaveBeenCalledWith(4);
  });
});