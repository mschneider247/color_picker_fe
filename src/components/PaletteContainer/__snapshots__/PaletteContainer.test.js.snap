// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteContainer component PalettesContainer should match its snapshot 1`] = `
<section
  id="Section_PaletteContainer"
>
  <div
    className="colorBox"
    key="0"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <button
      className="lock_btn"
      onClick={[Function]}
      value={0}
    >
      ðŸ”“
    </button>
    <p
      className="hexCode"
    />
  </div>
  <div
    className="colorBox"
    key="1"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <button
      className="lock_btn"
      onClick={[Function]}
      value={1}
    >
      ðŸ”“
    </button>
    <p
      className="hexCode"
    />
  </div>
  <div
    className="colorBox"
    key="2"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <button
      className="lock_btn"
      onClick={[Function]}
      value={2}
    >
      ðŸ”“
    </button>
    <p
      className="hexCode"
    />
  </div>
  <div
    className="colorBox"
    key="3"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <button
      className="lock_btn"
      onClick={[Function]}
      value={3}
    >
      ðŸ”“
    </button>
    <p
      className="hexCode"
    />
  </div>
  <div
    className="colorBox"
    key="4"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <button
      className="lock_btn"
      onClick={[Function]}
      value={4}
    >
      ðŸ”“
    </button>
    <p
      className="hexCode"
    />
  </div>
</section>
`;
